### PR TITLE
webgpu: refactor and optimize reduce kernel

### PR DESCRIPTION
**Pull request by xhcao**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5830)

### CODE REVIEW

**Comment by xhcao** [comment]
@qjia7 @axinging @haoyunfeix @gyagp Please take a look, thank you.

File: tfjs-backend-webgpu/src/kernels/reduce_webgpu.ts [hunk 1]
```diff
@@ -41,32 +41,25 @@ export class ReduceProgram implements WebGPUProgram {
...
-    this.dispatch = computeDispatch(
-        this.dispatchLayout, this.outputShape, this.workGroupSize);
+    this.dispatchLayout = flatDispatchLayout(this.outputShape);
+    // A work group only outputs a data, so we transfer [1, 1, 1] to compute
+    // dispatch size.
+    this.dispatch =
+        computeDispatch(this.dispatchLayout, this.outputShape, [1, 1, 1]);
 
     this.reduceType = reduceType;
     this.shaderKey = `reduce_${reduceType}_${outputDtype}`;
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/reduce_webgpu.ts:51]: nit: Can you also remove `outputDtype` from shaderKey? It seems that `outputDtype` is useless.

**Review by qjia7 — APPROVED** [review]
LGTM with one nit.

**Review by xhcao — COMMENTED** [review]
Jiajia, I had fixed the nit, please take a look again,

**Review by qjia7 — APPROVED** [review]
LGTM, thanks.

**Comment by qjia7** [comment]
@pyu10055 @mattsoulanille Please help merge it. Thanks.

**Pull request merged**